controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -120,8 +124,24 @@
     public CompletableFuture<UpdateStreamStatus> sealStream(final String scope, final String stream) {
         Exceptions.checkNotNullOrEmpty(scope, "scope");
         Exceptions.checkNotNullOrEmpty(stream, "stream");
-        return streamMetadataTasks.sealStream(scope, stream, null)
-                .thenApplyAsync(status -> UpdateStreamStatus.newBuilder().setStatus(status).build(), executor);
+        return streamStore.getActiveSegments(scope, stream, null, executor).thenCompose(segments -> {
+            if (segments.isEmpty()) {
+                return CompletableFuture.completedFuture(ScaleResponse.newBuilder()
+                        .setStatus(ScaleResponse.ScaleStreamStatus.SUCCESS).build());
+            } else {
+                return streamMetadataTasks.manualScale(scope,
+                        stream,
+                        segments.stream().map(Segment::getNumber).collect(Collectors.toList()),
+                        Collections.emptyList(),
```

Comment:
I looked into the code, we are not doing this validation == matching new ranges to sealed ranges. 
we should do that validation esp for manual scale or we may leave the stream in an inconsistent state. created a new issue (#1479) and assigned to self. I will make sure that i special case the check if segmentsToSeal is non empty and newRanges is empty.